File: resources/generate-icons.py
#!/usr/bin/env python3
"""
Generate icon files from SVG source.
Requires: pip install cairosvg pillow
"""

import subprocess
import sys
from pathlib import Path

def generate_icons():
    script_dir = Path(__file__).parent
    svg_path = script_dir / "icon.svg"
    
    if not svg_path.exists():
        print(f"Error: {svg_path} not found")
        sys.exit(1)
    
    print("Generating app icons from SVG...")
    
    try:
        # Try using cairosvg if available
        import cairosvg
        
        sizes = [256, 128, 96, 64, 48, 32, 16]
        
        # Generate PNG at 256x256
        print("  → Generating PNG (256x256)...")
        cairosvg.svg2png(
            url=str(svg_path),
            write_to=str(script_dir / "icon.png"),
            output_width=256,
            output_height=256
        )
        
        # Generate ICO with multiple sizes
        print("  → Generating ICO (multiple sizes)...")
        from PIL import Image
        
        png_path = script_dir / "icon.png"
        img = Image.open(png_path)
        
        ico_images = []
        for size in sizes:
            resized = img.resize((size, size), Image.Resampling.LANCZOS)
            ico_images.append(resized)
        
        ico_images[0].save(
            script_dir / "icon.ico",
            format="ICO",
            sizes=[(size, size) for size in sizes]
        )
        
        print("✅ Icons generated successfully!")
        print("   - icon.png (256x256)")
        print("   - icon.ico (multi-size)")
        print("\n⚠️  For macOS, you still need to generate icon.icns")
        print("   See ICON_SETUP.md for instructions")
        
    except ImportError as e:
        print("Error: Required packages not installed")
        print("Install with: pip install cairosvg pillow")
        print(f"Specific error: {e}")
        sys.exit(1)

if __name__ == "__main__":
    generate_icons()

File: resources/generate-icons.js
#!/usr/bin/env node

/**
 * Generate placeholder PNG icons from SVG using Node.js
 * This is a simple fallback when Python/cairosvg is not available
 */

const fs = require('node:fs');
const path = require('node:path');

// Very simple PNG generator - creates a blue placeholder icon
function createSimplePNG() {
  // PNG header and IHDR chunk for 256x256 blue image
  const width = 256;
  const height = 256;
  
  // Create a simple blue gradient image using raw pixel data
  const canvas = Buffer.alloc(width * height * 4);
  
  // Fill with airplane icon colors (blue gradient)
  for (let i = 0; i < canvas.length; i += 4) {
    canvas[i] = 91;      // R - blue
    canvas[i + 1] = 155; // G 
    canvas[i + 2] = 213; // B
    canvas[i + 3] = 255; // A - opaque
  }
  
  // Note: Full PNG encoding is complex; this is a placeholder
  console.log('ℹ️  Simple PNG generation not fully supported');
  console.log('Please use: npm install -g electron-icon-generator');
  console.log('Then run: electron-icon-generator --input=icon.svg --output=.');
  
  return null;
}

// Try using sharp or manual PNG creation
try {
  const sharp = require('sharp');
  
  console.log('Generating icons with Sharp...');
  
  const svgPath = path.join(__dirname, 'icon.svg');
  const svgContent = fs.readFileSync(svgPath, 'utf8');
  
  // Generate PNG
  sharp(Buffer.from(svgContent))
    .png()
    .resize(256, 256, { fit: 'contain', background: { r: 255, g: 255, b: 255, alpha: 0 } })
    .toFile(path.join(__dirname, 'icon.png'))
    .then(() => console.log('✅ Generated icon.png'))
    .catch(err => console.error('Error generating PNG:', err));
    
} catch (e) {
  console.error('Error:', e.message ?? e);
  console.log('⚠️  Sharp not available. Please install dependencies:');
  console.log('  npm install sharp');
  console.log('');
  console.log('Or use the Python generator:');
  console.log('  cd resources');
  console.log('  python -m pip install cairosvg pillow');
  console.log('  python generate-icons.py');
  console.log('');
  console.log('Or use online tool: https://icoconvert.com/');
}
